main.py

```diff
@@ -1,13 +1,20 @@
 import pymongo
 from pymongo.results import InsertOneResult
 from pymongo import MongoClient
+from fastapi import FastAPI
+import settings
+from datetime import datetime, timedelta
 from fastapi import FastAPI, Request
 from src.models import ProductBase
-from src.settings import MONGO_URI, MONGO_DB, PRODUCTS_COLLECTION
+from src.settings import MONGO_URI, MONGO_DB, PRODUCTS_COLLECTION, CURRENT_PERIOD_MONTHS, PREVIOUS_PERIOD_MONTHS
```

Comment:
Above you already have `settings` imported. Do you really need to reimport all the constants from `settings`?